pkg/kritis/admission/admission.go

```diff
@@ -18,45 +18,149 @@ package admission
 
 import (
 	"encoding/json"
-	"log"
-	"net/http"
-
+	"fmt"
 	"github.com/grafeas/kritis/pkg/kritis/admission/constants"
+	kritisv1beta1 "github.com/grafeas/kritis/pkg/kritis/apis/kritis/v1beta1"
+	"github.com/grafeas/kritis/pkg/kritis/crd/securitypolicy"
+	"github.com/grafeas/kritis/pkg/kritis/kubectl/plugins/resolve"
+	"github.com/grafeas/kritis/pkg/kritis/metadata"
+	"github.com/grafeas/kritis/pkg/kritis/metadata/containeranalysis"
+	"github.com/grafeas/kritis/pkg/kritis/pods"
+	"io/ioutil"
 	"k8s.io/api/admission/v1beta1"
+	"k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"log"
+	"net/http"
 )
 
-// This is very basic admission controller which allows all pods.
+type config struct {
+	retrievePod           func(r *http.Request) (*v1.Pod, error)
```

Comment:
These would be easier to mock if you take the pod name instead of the http request

Reply:
sorry, I'm not sure how that would change mocking out the function? since I wouldn't be using the http request or the pod name, and just returning a pod.